### ISSUE TITLE
Geany freezes desktop environment after using CTRL+K

### ISSUE THREAD

**Issue by toxpal** [geany/geany#2201]
Had exactly the same problem on antergos KDE distro, and have it on Manjaro KDE too.

Every time I use CTRL+K shortcut, Geany freezes desktop environment for 2-3 seconds. Everything is working perfectly in Geany itself (I can type text and do anything I want), but if I try to switch to another app from taskbar (or open Start menu) everything is frozen and I need to wait 2-3 seconds. Then desktop environment starts working again (until next CTRL+K in Geany).

It only happens with existing (saved) files and doesn't happen if I create new file, start typing and then use CTRL+K without saving file first.

Tried disabling all plugins, didn't help. Am I the only who has this issue?

`14:53:57: GLib-GIO DEBUG	: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
14:53:57: Geany INFO		: Geany 1.35, en_US.UTF-8
14:53:57: Geany INFO		: GTK 3.24.8, GLib 2.60.3
14:53:57: Geany INFO		: System data dir: /usr/share/geany
14:53:57: Geany INFO		: User config dir: /home/toxpal/.config/geany
14:53:57: Geany INFO		: Loaded GTK+ CSS theme '/usr/share/geany/geany.css'
14:53:57: Geany INFO		: Loaded GTK+ CSS theme '/usr/share/geany/geany-3.20.css'
14:53:57: Geany INFO		: System plugin path: /usr/lib/geany
14:53:57: Geany INFO		: Added filetype JSON (61).
14:53:57: Geany INFO		: Added filetype Graphviz (62).
14:53:57: Geany INFO		: Added filetype Cython (63).
14:53:57: Geany INFO		: Added filetype Clojure (64).
14:53:57: Geany INFO		: Added filetype Scala (65).
14:53:57: Geany INFO		: Added filetype Genie (66).
14:53:57: Geany INFO		: Added filetype CUDA (67).
14:53:57: Geany INFO		: Added filetype Arduino (68).
14:53:57: Geany INFO		: Disabling terminal support
14:53:57: Geany INFO		: Loaded:   /usr/lib/geany/addons.so (Addons)
14:53:57: Geany INFO		: Loaded:   /usr/lib/geany/automark.so (Auto-mark)
14:53:57: Geany INFO		: Loaded:   /usr/lib/geany/geanydoc.so (Doc)
14:53:57: Geany INFO		: Loaded:   /usr/lib/geany/pairtaghighlighter.so (Pair Tag Highlighter)
14:53:57: SpellCheck DEBUG	: Initializing Enchant library version 2.2.3
14:53:57: Geany INFO		: Loaded:   /usr/lib/geany/spellcheck.so (Spell Check)
14:53:57: Geany INFO		: unknown : None (UTF-8)
14:53:57: Geany INFO		: Bad color 'purewhitebluegrey'
14:53:57: Geany INFO		: Bad color '3'
`

**Comment by elextr** [comment]
Doesn't happen on Mint Cinnamon with Git Geany.

**Comment by elextr** [comment]
Can you try it without any of the plugins enabled and a default config, start Geany with `geany -c /tmp/some_thing_that_does_not_exist`

**Comment by toxpal** [comment]
Tested with default config and no plugins. Started Geany using the command you wrote.

Worked well after typing some short text, but then I pasted a content of PHP file (1500+ lines), used CTRL+L and my desktop frozen for a few seconds. As usually, Geany didn't freeze itself, it continued to work well.

**Comment by elextr** [comment]
Still can't reproduce on Mint cinnamon, would suspect its maybe some unfortunate interaction of GTK and KDE.

**Comment by elextr** [comment]
Also, what have you bound ctrl+l to?  By default its goto line, not a paste.

**Comment by toxpal** [comment]
I typed CTRL+L instead of CTRL+K by mistake when posting message above, and I quickly edited it.

**Comment by elextr** [comment]
> I typed CTRL+L instead of CTRL+K by mistake

Oh, a ytpo, no problem :)

**Comment by b4n** [comment]
what happens if you use another clipboard-related command, like <kbd>Ctrl+C</kbd> or <kbd>Ctrl+X</kbd>?

**Comment by toxpal** [comment]
> what happens if you use another clipboard-related command, like Ctrl+C or Ctrl+X?

Nothing, everything works perfectly :)

**Comment by b4n** [comment]
That's real odd… and actually by default <kbd>Ctrl+k</kbd> only *deletes* one or more lines, so it's not even related to clipboard.  @toxpal could you try changing the key combination for the action (*Edit → Preferences → Keybindings → Editor → Delete current line(s)*) and see if it still happens with the new binding?
My only guess for the moment would be that <kbd>Ctrl+k</kbd> is doing something at your desktop level as well.

**Comment by toxpal** [comment]
Yes, it acts exactly the same way (freezes for a few seconds). Used CTRL+Q (overwrote existing binding for "Quit" command).

**Comment by toxpal** [comment]
Did some more tests, and result is always the same - freeze after deleting line via shortcut (no matter which key binding was used) in both "normal" and "default config without plugins" modes. But it only happens when editing existing (saved) file. If I create a new file, paste thousands of lines into it and use CTRL + K, everything works until the moment I save the file on disk.

For this reason, I believe it might be somehow related to one file monitoring options (because Geany has plenty of them that are used to monitor files for external changes, do auto reload, etc)

**Comment by elextr** [comment]
> I believe it might be somehow related to one of file monitoring options

That possibly could freeze Geany, but not the rest of the desktop, and its just a `stat`, so unless that triggers the desktop to do a re-scan maybe ...

And monitoring isn't triggered by delete.

**Comment by elextr** [comment]
Can you list _exactly_ the steps you take to trigger it, including moving the cursor to check the desktop is working, moving between applications, switching Geany tabs everything.

**Comment by elextr** [comment]
I'll dump the partly formed theory here now because I'm gonna be AFK for a while.

Google shows a number of "freezes my KDE desktop" results blamed on the scanning component.

Some versions of Linux have a somewhat dodgy `notify` feature and so maybe the monitoring component is seeing a Geany `stat` as a metadata change and triggering a rescan, freezing the desktop.

Although the Geany file monitor `stat` isn't triggered by delete, it may be triggered by a tab switch or focus switch just before the delete, which is why the question about the exact sequence.

**Comment by toxpal** [comment]
Very basic steps...

1. Open existing file (better a bigger one, with 100+ lines at least);
2. Move mouse cursor over any line and click it to enter selected line;
3. Use CTRL+K
4. Move mouse cursor over any opened app in taskbar and click it - nothing will happen.
OR
4.1. After using CTRL+K, simply press "Start" button in keyboard (that one with Windows logo usually) and do nothing. Start menu will only open after ~3 seconds (after freeze caused by CTRL+K is gone).
